pkg/prometheus/server/operator.go

```diff
@@ -964,6 +972,44 @@ func (c *Operator) sync(ctx context.Context, key string) error {
 	return nil
 }
 
+// add or remove finalizers for the Prometheus resource.
+func (c *Operator) syncFinalizers(ctx context.Context, p *monitoringv1.Prometheus, key string) (bool, error) {
+	if c.configResourcesStatusEnabled {
+		var finalizersChanged bool
+		if !c.rr.DeletionInProgress(p) {
+			// Add finalizer to the Prometheus resource if it doesn't have one.
+			finalizers := p.GetFinalizers()
+			patchBytes, err := k8sutil.AddStatusCleanupFinalizer(finalizers)
+			if err != nil {
+				return false, fmt.Errorf("failed to marshal patch: %w", err)
+			}
+			if patchBytes != nil {
```

Comment:
(nit) we usually check on the length of the byte slice.
```suggestion
			if len(patchBytes) > 0 {
```